tools/automate_bindings.py

```diff
@@ -0,0 +1,321 @@
+    'int' : 'lua_pushinteger',
+    'bool' : 'lua_pushboolean',
+}
+
+def parse(filename):
+    # Find out the c++ parser.
+    generator_path, generator_name = pygccxml.utils.find_xml_generator()
+    # Configure the xml generator.
+    xml_generator_config = pygccxml.parser.xml_generator_configuration_t(
+        xml_generator_path = generator_path,
+        xml_generator = generator_name,
+    )
+    # Parse the c++ file.
+    decls = pygccxml.parser.parse([filename], xml_generator_config)
+    # Get access to the global namespace.
+    global_namespace = pygccxml.declarations.get_global_namespace(decls)
+    return global_namespace
+
+def getMethodsAndBases(global_namespace):
+    Wt = global_namespace.namespace('Wt')
+    methods = Wt.member_functions()
+    main_class = Wt.classes()[0]
+    bases = main_class.bases
+    return methods, bases
+
+def getConstructor(global_namespace):
+    Wt = global_namespace.namespace('Wt')
+    main_class = Wt.classes()[0]
+    # TODO (for zer0main).
+    # We need to support multiple constructors so it's just a dummy.
+    return pygccxml.declarations.find_trivial_constructor(main_class)
+
+def getModuleName(filename):
+    return os.path.basename(filename)
+
+INCLUDES_TEMPLATE = r'''
+#include "boost-xtime.hpp"
+#include <Wt/%s>
+
+#include "globals.hpp"
+
+'''
+
+def generateIncludes(module_name):
+    return INCLUDES_TEMPLATE.lstrip() % module_name
+
+def getSelf(module_name):
+    frame = r'''
+    %s* self = luawt_checkFromLua<%s>(L, 1);
+    '''
+    return frame % (module_name, module_name)
+
+def getInbuiltTypeArgument(options):
+    frame = r'''
+    %(argument_type)s %(argument_name)s = %(func)s(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def getComplexArgument(options):
+    frame = r'''
+    %(argument_type)s* %(argument_name)s =
+        luawt_checkFromLua<%(argument_type)s>(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def callWtConstructor(return_type, args, module_name):
+    call_s = 'new %s(' % module_name
+    args_s = ', '.join(arg.name for arg in args)
+    constructor_s = call_s + args_s + ');'
+    return '%s result = %s' % (return_type, constructor_s)
+
+def callWtFunction(return_type, args, method_name):
+    call_s = 'self->%s(' % method_name
+    args_s = ', '.join(arg.name for arg in args)
+    func_s = call_s + args_s + ');'
+    if return_type == 'void':
+        return func_s
+    else:
+        return '%s result = %s' % (return_type, func_s)
+
+RETURN_CALLS_TEMPLATE = r'''
+    %s(L, result);
+    return 1;
+'''
+
+def returnValue(return_type):
+    void_frame = r'''
+    return 0;
+    '''
+    if return_type == 'void':
+        return void_frame
+    else:
+        inbuilt_type = getInbuiltType(return_type)
+        if inbuilt_type:
+            func_name = TYPE_TO_LUA_FUNCS[inbuilt_type]
+        else:
+            func_name = 'luawt_toLua'
+        return RETURN_CALLS_TEMPLATE % func_name
+
+def getInbuiltType(full_type):
```

Comment:
There is no word builtin (without dash) in English. Dash is totally unacceptable for style reasons.
The word `inbuilt` exists is and it's quite good for this situation.